src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethodsTests.cs

```diff
@@ -376,6 +376,151 @@ End Function
 End Class");
         }
 
+        [Fact, WorkItem(2914, "https://github.com/dotnet/roslyn-analyzers/issues/2914")]
+        public void CA1721_OverrideNoDiagnosticButVirtualDiagnostic()
+        {
+            VerifyCSharp(@"
+public class BaseClass
+{
+    public virtual int Value { get; }
+    public virtual int GetValue(int i) => i;
+}
+
+public class C1 : BaseClass
+{
+    public override int Value => 42;
+}
+
+public class C2 : BaseClass
+{
+    public override int GetValue(int i) => i * 2;
+}
+
+public class C3 : BaseClass
+{
+    public override int Value => 42;
+    public override int GetValue(int i) => i * 2;
+}
+",
+            GetCA1721CSharpResultAt(line: 4, column: 24, identifierName: "Value", otherIdentifierName: "GetValue"));
+
+            VerifyBasic(@"
+Public Class BaseClass
+    Public Overridable ReadOnly Property Value As Integer
+
+    Public Overridable Function GetValue(ByVal i As Integer) As Integer
+        Return i
+    End Function
+End Class
+
+Public Class C1
+    Inherits BaseClass
+
+    Public Overrides ReadOnly Property Value As Integer
+        Get
+            Return 42
+        End Get
+    End Property
+End Class
+
+Public Class C2
+    Inherits BaseClass
+
+    Public Overrides Function GetValue(ByVal i As Integer) As Integer
+        Return i * 2
+    End Function
+End Class
+
+Public Class C3
+    Inherits BaseClass
+
+    Public Overrides ReadOnly Property Value As Integer
+        Get
+            Return 42
+        End Get
+    End Property
+
+    Public Overrides Function GetValue(ByVal i As Integer) As Integer
+        Return i * 2
+    End Function
+End Class
+",
+        GetCA1721BasicResultAt(line: 3, column: 42, identifierName: "Value", otherIdentifierName: "GetValue"));
+        }
+
+        [Fact, WorkItem(2914, "https://github.com/dotnet/roslyn-analyzers/issues/2914")]
+        public void CA1721_OverrideWithLocalMemberDiagnostic()
+        {
+            VerifyCSharp(@"
+public class BaseClass1
+{
+    public virtual int Value { get; }
```

Comment:
Can you add a test where the virtual in base type is a method instead of a property?

Reply:
See tests from line 466+. `BaseClass2` has a virtual method and `C2` adds the property.